client/modules/IDE/components/Header/MobileNav.jsx

```diff
@@ -0,0 +1,388 @@
+  const user = useSelector((state) => state.user);
+  const dispatch = useDispatch();
+
+  // use the useNavigate hook in the react router v6
+  const history = useHistory();
+  const navigate = (url) => history.push(url);
+
+  return (
+    <div>
+      <IconButton icon={AccountIcon} />
+      <ul>
+        <li className="user">{user.username}</li>
+        <li>
+          <button
+            onClick={() => {
+              navigate(`/${user.username}/sketches`);
+            }}
+          >
+            My Stuff
+          </button>
+        </li>
+        <li>
+          <button
+            onClick={() => {
+              navigate(`/account`);
+            }}
+          >
+            Settings
+          </button>
+        </li>
+        <li>
+          <button onClick={() => dispatch(logoutUser())}>Log Out</button>
+        </li>
+      </ul>
+    </div>
+  );
+};
+
+const MoreMenu = () => {
+  // TODO: use selectRootFile selector
+  const rootFile = useSelector(
+    (state) => state.files.filter((file) => file.name === 'root')[0]
+  );
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const { newSketch, saveSketch } = useSketchActions();
+  const history = useHistory();
+  const navigate = (url) => history.push(url);
+
+  const cmRef = useContext(CmControllerContext);
+
+  return (
+    <div>
+      <IconButton icon={MoreIcon} />
+      <ul>
+        <b>{t('Nav.File.Title')}</b>
+        <li>
+          <button onClick={newSketch}>{t('Nav.File.New')}</button>
+        </li>
+        <li>
+          <button onClick={() => saveSketch(cmRef.current)}>
+            {t('Common.Save')}
+          </button>
+        </li>
+        <li>
+          <button
+            onClick={() => {
+              navigate(`/p5/sketches`);
+            }}
+          >
+            {t('Nav.File.Examples')}
+          </button>
+        </li>
+        <b>{t('Nav.Edit.Title')}</b>
+        <li>
+          <button onClick={cmRef.current?.tidyCode}>
+            {t('Nav.Edit.TidyCode')}
+          </button>
+        </li>
+        <li>
+          <button onClick={cmRef.current?.showFind}>
+            {t('Nav.Edit.Find')}
+          </button>
+        </li>
+        <b>{t('Nav.Sketch.Title')}</b>
+        <li>
+          <button onClick={() => dispatch(newFile(rootFile.id))}>
+            {t('Nav.Sketch.AddFile')}
+          </button>
+        </li>
+        <li>
+          <button onClick={() => dispatch(newFolder(rootFile.id))}>
+            {t('Nav.Sketch.AddFolder')}
+          </button>
+        </li>
+        {/* TODO: Add Translations */}
+        <b>Settings</b>
+        <li>
+          <button
+            onClick={() => {
+              dispatch(openPreferences());
+            }}
+          >
+            Preferences
+          </button>
+        </li>
+        <li>
+          <button>Language</button>
+        </li>
+        <b>{t('Nav.Help.Title')}</b>
+        <li>
+          <button>{t('Nav.Help.KeyboardShortcuts')}</button>
```

Comment:
I think the one is just a `dispatch` to open the modal.

However — if there’s a way to hide this menu item on touch devices that would be super cool!  It would be a media query that’s specifically for whether it’s touch or keyboard, not based on the width.  The might have a keyboard plugged in to a small tablet, for example.

That’s for the future.  For now open the modal.

Reply:
okayy!